lib/pg_ha_migrations/unsafe_statements.rb

```diff
@@ -80,7 +80,15 @@ def unsafe_add_index(table, column_names, options = {})
       raise PgHaMigrations::InvalidMigrationError, "ActiveRecord drops the :opclass option when supplying a string containing an expression or list of columns; instead either supply an array of columns or include the opclass in the string for each column"
     end
 
-    execute_ancestor_statement(:add_index, table, column_names, **options)
+    if options[:algorithm] == :concurrently
+      # Cannot be run inside a transaction so we are unable to use
+      # safely_acquire_lock_for_table to take out ShareUpdateExclusiveLock
+      execute_ancestor_statement(:add_index, table, column_names, **options)
```

Comment:
This is an interesting catch. This is a project philosophy question, but should we insist that the user call `safe_add_index_concurrently`?

Reply:
Not sure I'm following this one

Reply:
This seems to be from an earlier version that updated where we did locking. Since that's no longer true in the PR, I think we can wait on this (but I also don't quite remember the context of my question about project policy, and I'm struggling to find the version of the PR at the time I left the comment, so I can't see the full context of this change...)